relay-server/src/services/processor.rs

```diff
@@ -2669,14 +2710,61 @@ mod tests {
     use crate::testutils::{self, create_test_processor};
 
     #[cfg(feature = "processing")]
-    use {
-        relay_metrics::BucketValue,
-        relay_quotas::{Quota, ReasonCode},
-        relay_test::mock_service,
-    };
+    use {relay_metrics::BucketValue, relay_quotas::ReasonCode, relay_test::mock_service};
 
     use super::*;
 
+    fn mock_quota(id: &str) -> Quota {
+        Quota {
+            id: Some(id.into()),
+            categories: smallvec::smallvec![DataCategory::MetricBucket],
+            scope: QuotaScope::Organization,
+            scope_id: None,
+            limit: Some(0),
+            window: None,
+            reason_code: None,
+            namespace: None,
+        }
+    }
+
+    #[test]
+    fn test_dynamic_quota_empty() {
+        let glob_config = GlobalConfig::default();
+        let dynq = DynamicQuotas::new(&glob_config, &[]);
+        assert!(dynq.is_empty());
+    }
+
+    #[test]
+    fn test_dynamic_quota_len() {
+        let global_config = GlobalConfig {
+            measurements: None,
+            quotas: vec![mock_quota("foo"), mock_quota("bar")],
+            options: Options::default(),
+        };
+
+        let quotas = vec![mock_quota("baz")];
+
+        let dynq = DynamicQuotas::new(&global_config, &quotas);
+        assert_eq!(dynq.len(), 3);
```

Comment:
I think it's fine if we just assert the `len()` and `is_empty()` methods in `test_dynamic_quotas`:

```rs
assert_eq!(foo.len(), 5);
assert!(!foo.is_empty());
```